tree/tree/src/TTree.cxx

```diff
@@ -8248,6 +8258,13 @@ void TTree::ResetBranchAddresses()
          }
       }
    }
+   if (GetListOfClones()) {
+      for (TObjLink *lnk = GetListOfClones()->FirstLink(); lnk; lnk = lnk->Next()) {
+         TTree *clone = (TTree *)lnk->GetObject();
+         if (clone)
+            clone->ResetBranchAddresses();
```

Comment:
If we just reset the addresses of the clone, the address are now valid but the current tree and the clone are still separated.  [As a side note calling `ResetBranchAddresses` actually does too much because it also reset branch address that are unrelated to the original, for example branches that have been added to the clone).
```suggestion
            CopyAddresses(clone);
```